### PR TITLE
doc: ci: workflow improvements and PDF support

### PR DESCRIPTION
**Pull request by gmarull**
This PR brings some improvements to the documentation workflow:

- A single "Documentation Build" workflow is used to build the documentation, regardless if it's for a PR, a release or a scheduled event.
- The "Documentation Publish" workflow will run on `main` or releases after documentation build succeeds. It re-uses artifacts from the build stage.
- PDF support has been added (build and publish)

NOTE: I'm not entirely familiar with how AWS S3 is configured for Zephyr, so that part may require more attention as I may have missed something.

### CODE REVIEW

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -63,3 +63,43 @@ jobs:
...
+        pip3 install -U setuptools wheel pip west
+        pip3 install -r scripts/requirements-doc.txt
+
+    - name: west setup
+      run: |
+        west init -l .
+
+    - name: build-docs
+      run: |
+        cmake -Sdoc -Bbuild -DSPHINXOPTS="-q -j auto" -DLATEXMKOPTS="-quiet -halt-on-error"
```
Comment (nashif) [inline .github/workflows/doc-build.yml:98]: why not call 'make pdf' to make sure our build process also works?

File: .github/workflows/doc-build.yml [hunk 2]
```diff
@@ -63,3 +63,43 @@ jobs:
...
+        cmake --build build --target pdf
+
+    - name: upload-build
+      uses: actions/upload-artifact@master
+      continue-on-error: True
+      with:
+        path: build/latex/zephyr.pdf
```
Comment (nashif) [inline .github/workflows/doc-build.yml:105]: add name here, so it is visible that this is the PDF.

```
name: pdf-output
```

for example.

Comment (gmarull) [reply .github/workflows/doc-build.yml:101]: added

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (gmarull) [reply .github/workflows/doc-build.yml]: added `LATEXMKOPTS` option to Makefile (required for CI), now CI uses "make pdf"

**Review by carlescufi — APPROVED** [review]
Looks good to me!

File: .github/workflows/doc-build.yml [hunk 3]
```diff
@@ -4,6 +4,11 @@
 name: Documentation GitHub Workflow
 
 on:
+  schedule:
```
Comment (nashif) [inline .github/workflows/doc-build.yml:7]: do not understand why this is needed.You need to explain why do we need another job building the docs on schedule in the main branch, we have this already.

File: .github/workflows/doc-publish.yml [hunk 4]
```diff
@@ -1,48 +1,35 @@
 # Copyright (c) 2020 Linaro Limited.
+# Copyright (c) 2021 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-name: Doc build for Release or Daily
+name: Documentation Publish
```
Comment (nashif) [inline .github/workflows/doc-publish.yml:5]: Publish Documentation

**Review by nashif — CHANGES_REQUESTED** [review]

File: .github/workflows/doc-build.yml (hunk 3 above)
Comment (gmarull) [reply .github/workflows/doc-build.yml:7]: The publish workflow has been refactored to use the workflow_run event. It doesn't build anymore, just publishes from imported artifacts. This way we have a single workflow that is responsible for building docs and so we remove duplicates. A working example: https://github.com/nrfconnect/sdk-nrf/blob/master/.github/workflows/docpublish.yml

File: .github/workflows/doc-build.yml [hunk 5]
```diff
@@ -25,10 +25,6 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Update PATH for west
-      run: |
-        echo "$HOME/.local/bin" >> $GITHUB_PATH
```
Comment (mbolivar-nordic) [inline .github/workflows/doc-build.yml:30]: I guess recent ubuntu is doing this for the user automatically? It was required on that OS at some point in the past.

**Review by mbolivar-nordic — APPROVED** [review]
Sure, no objections to this approach

File: .github/workflows/doc-build.yml (hunk 5 above)
Comment (gmarull) [reply .github/workflows/doc-build.yml:30]: This is the `PATH` of the `ubuntu-latest` runner:

```
PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/runner/.local/bin:/opt/pipx_bin:/usr/share/rust/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

so yes, `/home/runner/.local/bin` is already there.

File: .github/workflows/doc-publish.yml (hunk 4 above)
Comment (gmarull) [reply .github/workflows/doc-publish.yml]: adjusted

**Review by nashif — APPROVED** [review]

**Pull request merged**